Sources/MongoSwift/MongoClient.swift

```diff
@@ -173,8 +198,11 @@ public class MongoClient {
         self.readPreference = options.readPreference ?? ReadPreference(.primary)
         self.encoder = BSONEncoder(options: options)
         self.decoder = BSONDecoder(options: options)
+        self.notificationCenter = options.notificationCenter ?? NotificationCenter.default
```

Comment:
I wonder, if we're making more dramatic changes, whether we should get rid of storing a notification center at all. If we instead supported passing in lambdas, or some sort of protocol for callbacks, then the user could decide to use whatever means of notification they wanted (a notification center, some sort of sswg log facility, writing to a file, etc).

Reply:
yeah, we made a ticket about this a while back: [SWIFT-360](https://jira.mongodb.org/browse/SWIFT-360)

I definitely agree we should change this API to give the user a bit more flexibility about what happens. technically now they still have it, because the observer they register on the notification center can do anything they want it to. but we may as well just let them register a callback with us directly.
I think this should end up being kind of symmetric to the logging API SSWG introduced, and we should think about bringing in autoclosures so that if e.g. a user only cares about `CommandStartedEvent`s we don't waste time filling out structs for the other command-related events which will just get thrown away. (this may be tricky with the mongoc event object lifetimes though...)

I think this kind of needs a mini-design... it would be nice if we can get it into the same release as these changes, but I think we could switch to the callbacks in a non-breaking way in the future if we want - we could just default to still publishing to the notification center but allow you to override

Reply:
per conversation in triage today with @patrickfreed and @rathisekaran I'm going to merge this PR in as-is. I am marking SWIFT-360 with the next fix version so we remember to revisit it before releasing these new changes.